### PR TITLE
[next-auth] Add idToken option

### LINKED ISSUE #52018: [@types/next-auth] ProviderCommonOptions idToken missing
- [x] I tried using the `@types/next-auth` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I understand that `@types` package versions [do not follow SemVer](https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library).
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @luia @iaincollins @JuanM04 @balazsorban44 @euxn23

ProviderCommonOptions needs idToken prop because some providers are using id_token instead of access_token. Next-Auth have a conditional if on that prop. I need to set it for AzureADB2C provider.

https://github.com/nextauthjs/next-auth/blob/782482b9f475ca19019e5bc37f3053718d472cb2/src/server/lib/oauth/callback.js#L42

### PR DESCRIPTION
**Pull request by martin-badin**
Closes: #52018

Hi, I added the option but I am not sure if the place is correct. Thanks for the review.

Source file: https://github.com/nextauthjs/next-auth/blob/782482b9f475ca19019e5bc37f3053718d472cb2/src/server/lib/oauth/callback.js#L42

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/next-auth/next-auth-tests.ts [hunk 1]
```diff
@@ -563,6 +563,7 @@ Providers.AzureADB2C({
     clientSecret: "bar123",
     scope: "offline_access User.Read",
     tenantId: "tenantId",
+    idToken: "idToken",
```
Comment (balazsorban44) [inline types/next-auth/next-auth-tests.ts:566]: this should be a boolean, not a string

File: types/next-auth/providers.d.ts [hunk 2]
```diff
@@ -123,9 +123,10 @@ type OptionsBase = {
 };
 
 interface ProviderCommonOptions extends OptionsBase {
-    name?: string;
```
Comment (balazsorban44) [inline types/next-auth/providers.d.ts:126]: why did you move this?

File: types/next-auth/providers.d.ts [hunk 3]
```diff
@@ -123,9 +123,10 @@ type OptionsBase = {
...
     clientId: string;
     clientSecret: string;
+    idToken?: string;
```
Comment (balazsorban44) [inline types/next-auth/providers.d.ts:128]: https://next-auth.js.org/configuration/providers#oauth-provider-options

**Review by balazsorban44 — CHANGES_REQUESTED** [review]

File: types/next-auth/providers.d.ts (hunk 2 above)
Comment (martin-badin) [reply types/next-auth/providers.d.ts:126]: I sorted the options alphabetically.

File: types/next-auth/providers.d.ts (hunk 3 above)
Comment (martin-badin) [reply types/next-auth/providers.d.ts:128]: thanks, I'll add all missing options

Comment (martin-badin) [reply types/next-auth/providers.d.ts]: updated

File: types/next-auth/next-auth-tests.ts (hunk 1 above)
Comment (martin-badin) [reply types/next-auth/next-auth-tests.ts]: Yes, you are right. updated

**Review by balazsorban44 — APPROVED** [review]

**Comment by martin-badin** [comment]
Ready to merge
